### PR TITLE
(#4339) - Failing get after destroy test

### PR DESCRIPTION
**Pull request by nickcolley**

### CODE REVIEW

File: lib/adapters/leveldb/index.js [hunk 1]
```diff
@@ -342,6 +342,7 @@ function LevelPouch(opts, callback) {
     opts = utils.clone(opts);
 
     stores.docStore.get(id, function (err, metadata) {
+      console.log(id);
```
Comment (gr2m) [inline lib/adapters/leveldb/index.js]: don't forget to remove this one again ;)

Comment (nickcolley) [inline lib/adapters/leveldb/index.js]: left it in so you could see that it never returns

**Comment by daleharvey** [comment]
Yup this has come up before, the behaviour this is testing was never one that we have catered for, it would have sometimes worked (and sometimes not) n the http adapter, I think almost always break in any of the other adapters, when you call `db.destroy` we generally expect that to be the end of the life of the object, if you need another db handle you need to reopen it.

We can wipe out the handle when `destroy` is called, make it throw an obvious error if someone calls any further functions on it, we could also support this use case (being able to reuse the handle after `destroy` by making all the constructor stateless.

**Comment by nickcolley** [comment]
It fails silently at the moment so no error message :(

**Comment by nolanlawson** [comment]
Maybe we should put a `help wanted` on this and open up an issue? Should be a pretty straightforward fix to have the promise reject with a helpful error message.

**Comment by nolanlawson** [comment]
Since nobody seems to be attacking this, I'm going to take it to mean that everybody is happy with the workaround where you just don't try to re-use a destroyed database.

**Pull request closed without merging**